docs/testplan/platform/pmon-test-plan.md

```diff
@@ -1,113 +1,227 @@
-* Verify the "PCIE_DEVICES" has "status" field set to "SUCCESS" 
-
-### Test case # 5 – Data for psud daemon
-#### Test objective
-Verify the expected data for psud daemon (if not skip_psud)
-#### Test steps
-* Verify the 'PSU_INFO|PSU {}' is/are available
-* Verify the information data for each 'PSU {}' is expected to compare "/usr/share/sonic/platform/platform.json"
-
-### Test case # 6 – Data for syseepromd daemon
-#### Test objective
-Verify the expected data for syseepromd daemon (if not skip_syseepromd)
-#### Test steps
-* Verify the 'EEPROM_INFO|State' in state_db has the 'Initialized' field set to '1'
-* Verify the 'EEPROM_INFO|*' has valid pairs of key and value
-   * Verify the data to check with the output of the platform api 
-
-### Test case # 7 – Data for thermalctld daemon
-#### Test objective
-Verify the expected data for thermalctld daemon (if not skip_thermalctld)
-#### Test steps
-* Verify the policy file "/usr/share/sonic/platform/thermal_policy.json" exists
-* Verify the 'FAN_INFO|{}' and 'TEMPERATURE_INFO|{}' available
-   * Verify the data to check with the output of the platform api 
-
-### Test case # 8 – Data for xcvrd daemon
-#### Test objective
-Verify the expected data for xcvrd daemon (if not skip_xcvrd)
-#### Test steps
-* Verify the 'TRANSCEIVER_INFO', 'TRANSCEIVER_DOM_SENSOR' and 'TRANSCEIVER_STATUS' state_db table has data
-* Verify the information data for each 'TRANSCEIVER_INFO' is expected to compare "/usr/share/sonic/platform/platform.json"
-   * Verify the data to check with the output of the platform api 
-
-## TODO
-
-## Open questions
+### Test steps
+1. Parse /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2,
+   /usr/share/sonic/platform/pmon_daemon_control.json, and /etc/supervisor/conf.d/supervisord.conf
+2. Find the list of daemons running in PMON docker for the specific platform of DUT
+   a. Check all available daemons in SONiC image
+      - Check the output of "sudo docker exec pmon bash -c '[ -f /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 ] \
+      && cat /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 | grep program'"
+      - Add the daemons from the output as "all_daemons" list
+      ```
+   admin@sonic:~$ sudo docker exec pmon bash -c '[ -f /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 ]  && cat /usr/share/sonic/templates/docker-pmon.supervisord.conf.j2 | grep program'
+   [program:rsyslogd]
+   [program:chassisd]
+   [program:lm-sensors]
+   [program:fancontrol]
+   [program:ledd]
+   [program:pcied]
+   [program:psud]
+   [program:syseepromd]
+   [program:thermalctld]
+   [program:xcvrd]
+      ```
+   b. Find any daemon that doesn't support by the platform 
+      - Check the output of "sudo docker exec pmon bash -c '[ -f /usr/share/sonic/platform/pmon_daemon_control.json ] \
+      && cat /usr/share/sonic/platform/pmon_daemon_control.json'"
+      - Add the skip_*** as "skipped_daemons" list
```

Comment:
Suggest to reword:

- Add each `skip_<daemon_name>` entry to a "skipped_daemons" list